scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
```

Comment:
You can restructure the existing linprog tests.

Since there are quite many of them, the following approach is probably simplest:
```
class LinprogCommonTests(object):
    ...

    def test_enzo_example_c_with_degeneracy(self):
        ...
        res = linprog(c=c, A_eq=A_eq, b_eq=b_eq, method=self.method)
        ...

    ...


class TestLingprogSimplex(LinprogCommonTests):
    method = "simplex"


class TestLingprogIP(LinprogCommonTests):
    method = "ip"

```
Tests that are not common for the two solvers can be left alone.